misc-scripts/distribution.py

```diff
@@ -0,0 +1,140 @@
+    print(f"--------- withdraw-rewards of {FROMKEY} -----------")
+    
+    wrcTx = f"{DAEMON} tx distribution withdraw-rewards {VALADDRESS} --from {FROMKEY} --commission --fees 1000{DENOM} --chain-id {CHAINID} --keyring-backend test --home {DAEMON_HOME}-{a} --node {RPC} --output json -y"
+    process = subprocess.Popen(wrcTx.split(), stdout=subprocess.PIPE, text=True)
+    output, error = process.communicate()
+    time.sleep(6)
+    res = json.loads(output)
+    wrctxHash = res["txhash"]
+    print(f"** TX HASH :: {wrctxHash} **")
+    txResult = f"{DAEMON} q tx {wrctxHash} --node {RPC} --output json"
+    process = subprocess.Popen(txResult.split(), stdout=subprocess.PIPE, text=True)
+    output, error = process.communicate()
+    res = json.loads(output)
+    wrcCode = res["code"]
+    reason = res["raw_log"]
+    if wrcCode == 0:
+        print(f"**** withdraw-rewards commission of {VALADDRESS}and key {FROMKEY} is successfull!! txHash is : {wrctxHash} *****")
+    else:
+        print(f"**** withdraw-rewards commission of {VALADDRESS} and key {FROMKEY} is failed!!! txHash is : {wrctxHash} and REASON: {reason} ****")
+    
+print("--------- Run withdraw-all-rewards tx -----------")
+
+for a in range(1,args.nodes+1):
+    DIFF = a-1
+    INC = DIFF*2
+    PORT = 16657 + INC
+    RPC = f"http://{IP}:{PORT}"
+    print(f"**** NODE :: {RPC} ******")
+    validator = f"{DAEMON} keys show validator{a} --bech val --keyring-backend test --home {DAEMON_HOME}-{a} --output json"
+    process = subprocess.Popen(validator.split(), stdout=subprocess.PIPE, text=True)
+    output, error = process.communicate()
+    res = (json.loads(output))
+    VALADDRESS = res["address"]
+    FROMKEY = f"validator{a}"
+    print(f"** validator address :: {VALADDRESS} and From key :: {FROMKEY} **")
+    print(f"Iteration no {a} and values of address : {VALADDRESS} and key : {FROMKEY}")
+    print(f"--------- withdraw-all-rewards of {FROMKEY} -----------")
+    warTx = f"{DAEMON} tx distribution withdraw-all-rewards --from {FROMKEY} --fees 1000{DENOM} --chain-id {CHAINID} --keyring-backend test --home {DAEMON_HOME}-{a} --node {RPC} --output json -y"
+    process = subprocess.Popen(warTx.split(), stdout=subprocess.PIPE, text=True)
+    output, error = process.communicate()
+    time.sleep(6)
+    res = json.loads(output)
+    wartxHash = res["txhash"]
+    print(f"** TX HASH :: {wartxHash}**")
+    txResult = f"{DAEMON} q tx {wartxHash} --node {RPC} --output json"
+    process = subprocess.Popen(txResult.split(), stdout=subprocess.PIPE, text=True)
+    output, error = process.communicate()
+    res = json.loads(output)
+    warcode = res["code"]
+    reason = res["raw_log"]
+    if warcode == 0:
+        print(f"**** withdraw-all-rewards of {VALADDRESS} and key {FROMKEY} is successfull!! txHash is : {wartxHash} *****")
+    else:
+        print(f"**** withdraw-all-rewards of {VALADDRESS} and key {FROMKEY} is failed!!! txHash is : {wartxHash} and REASON : {reason} ****")
+
```

Comment:
Lines 136-139: Need not print the errors. I assume the `subprocess.run[command]` itself will print the error. (Check whether the command failed status is printing twice or not.)